addons/im_livechat/models/discuss_channel.py

```diff
@@ -610,6 +611,52 @@ def _get_channel_history(self):
     def _get_livechat_session_fields_to_store(self):
         return []
 
+    def _apply_livechat_feedback(self, rate, reason=None):
+        """Apply customer feedback rating to a live chat session.
+        Creates a new rating record if none exists, or updates the existing one.
+        Posts a notification message in the channel with the rating details.
+
+        :param rate: rating value
+        :param reason: feedback text received from the customer
+        :return: ID of the created or updated rating record
+        """
+        self.ensure_one()
+        # sudo: rating.rating - visitor can access rating to check if feedback was already given
+        if not self.sudo().rating_ids:
+            partner, guest = self.env["res.partner"]._get_current_persona()
+            # sudo: rating.rating - live chat customers can create ratings
+            self.env["rating.rating"].sudo().create({
+                "rating": rate,
+                "consumed": True,
+                "feedback": reason,
+                "is_internal": False,
+                "res_id": self.id,
+                "res_model_id": self.env["ir.model"]._get_id("discuss.channel"),
+                # sudo: res.partner - live chat customers can set the related_partner_id
+                "rated_partner_id": self.sudo().channel_partner_ids[:1].id,
+                "partner_id": partner.id if partner else (guest.id if guest else False),
```

Comment:
```suggestion
                "partner_id": partner.id or guest.id,
```
:pick: `_get_current_persona` returns an empty recordset.